framework/include/utils/PetscSupport.h

```diff
@@ -38,14 +38,23 @@ namespace PetscSupport
 class PetscOptions
 {
 public:
-  PetscOptions() : flags("", "", true) {}
+  PetscOptions()
+    : flags("", "", true), ignored_options("", "", true), user_set_options("", "", true)
+  {
+  }
 
   /// PETSc key-value pairs
   std::vector<std::pair<std::string, std::string>> pairs;
 
   /// Single value PETSc options (flags)
   MultiMooseEnum flags;
 
+  /// Flags to explicitly not set, even if they are specified in flags
+  MultiMooseEnum ignored_options;
```

Comment:
I would prefer that we generally replace `ignored` with something like `dont_add_these`. Ignoring something scares me. It sounds like someone has asked for something but we're going to ignore it. "Don't add this" sounds a lot less scary and also reflects more truly what your code is doing: it's making it so that we don't add particular flags that don't make sense for a given simulation as opposed to "we're going to add these flags that don't make sense and then ignore them"